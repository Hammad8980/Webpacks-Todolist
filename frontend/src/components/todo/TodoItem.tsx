import { useState } from 'react';
import type { PropsWithChildren } from 'react';
import type { Task } from '../../features/todos/TodoTaskTypes';
import Button from '../ui/Button';

type DeletePhase =
  | 'idle'
  | 'expanding'
  | 'spreading'
  | 'transforming'
  | 'fading'
  | 'complete';

type TodoItemProps = PropsWithChildren<{
  task: Task;
  onDelete: (id: number) => void;
  onToggle: (id: number) => void;
  onUpdate: (id: number, title: string) => void;
}>;

export default function TodoItem({
  task,
  children,
  onDelete,
  onToggle,
  onUpdate,
}: TodoItemProps) {
  const [isEditing, setIsEditing] = useState(false);
  const [editTitle, setEditTitle] = useState(task.title);
  const [deletePhase, setDeletePhase] = useState<DeletePhase>('idle');

  const handleSave = () => {
    if (editTitle.trim() !== '') {
      onUpdate(task.id, editTitle.trim());
      setIsEditing(false);
    }
  };

  const handleCancel = () => {
    setEditTitle(task.title);
    setIsEditing(false);
  };

  const handleDelete = () => {
    if (deletePhase !== 'idle') return;

    setDeletePhase('expanding');

    // Phase 1: Expand red background (0.3s)
    setTimeout(() => {
      setDeletePhase('spreading');
    }, 300);

    // Phase 2: Continue expanding and start icon transformation (0.2s)
    setTimeout(() => {
      setDeletePhase('transforming');
    }, 500);

    // Phase 3: Show checkmark and prepare for fade (0.3s)
    setTimeout(() => {
      setDeletePhase('fading');
    }, 800);

    // Phase 4: Complete fade out and remove (0.4s)
    setTimeout(() => {
      setDeletePhase('complete');
      onDelete(task.id);
    }, 1200);
  };

  return (
    <>
      <div
        className={`relative flex items-center gap-3 p-4 rounded-xl border transition-all duration-200 shadow-sm hover:shadow-md overflow-hidden ${
          deletePhase === 'fading'
            ? 'animate-fadeOut'
            : deletePhase === 'idle'
              ? 'bg-gray-700 hover:bg-gray-650 border-gray-600'
              : 'bg-gray-700 border-gray-600'
        }`}
      >
        {/* Red expanding background */}
        <div
          className={`absolute inset-0 bg-red-600 transition-all duration-500 ease-out ${
            deletePhase === 'expanding'
              ? 'scale-x-50 opacity-80'
              : deletePhase === 'spreading'
                ? 'scale-x-100 opacity-90'
                : deletePhase === 'transforming' || deletePhase === 'fading'
                  ? 'scale-100 opacity-95'
                  : 'scale-x-0 opacity-0'
          }`}
          style={{
            transformOrigin: 'right center',
            borderRadius: 'inherit',
          }}
        />

        {/* Content layer */}
        <div className="relative z-10 flex items-center gap-3 w-full">
          <input
            type="checkbox"
            checked={task.isCompleted}
            onChange={() => onToggle(task.id)}
            className="h-5 w-5 text-blue-500 bg-gray-600 border-gray-500 rounded focus:ring-blue-400 focus:ring-2"
            disabled={deletePhase !== 'idle'}
          />
          <span
            className={`flex-grow text-lg transition-colors duration-200 text-justify break-all ${
              task.isCompleted ? 'line-through text-gray-400' : 'text-gray-200'
            } ${deletePhase !== 'idle' ? 'text-white' : ''}`}
          >
            {task.title}
          </span>
          <div className="flex gap-2">
            <Button
              onClick={() => setIsEditing(true)}
              disabled={deletePhase !== 'idle'}
              className="bg-blue-600 hover:bg-blue-700 text-white p-2 rounded-lg transition-all duration-200 shadow-sm hover:shadow-md disabled:opacity-50 disabled:cursor-not-allowed"
            >
              <svg
                className="w-4 h-4"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
              >
                <path
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth={2}
                  d="M11 5H6a2 2 0 00-2 2v11a2 2 0 002 2h11a2 2 0 002-2v-5m-1.414-9.414a2 2 0 112.828 2.828L11.828 15H9v-2.828l8.586-8.586z"
                />
              </svg>
            </Button>

            {/* Advanced Delete Button */}
            <Button
              onClick={handleDelete}
              disabled={deletePhase !== 'idle'}
              className={`relative p-2 rounded-lg transition-all duration-200 shadow-sm hover:shadow-md disabled:cursor-not-allowed overflow-hidden ${
                deletePhase === 'expanding' ||
                deletePhase === 'spreading' ||
                deletePhase === 'transforming'
                  ? 'bg-red-700 scale-110'
                  : deletePhase === 'idle'
                    ? 'bg-red-600 hover:bg-red-700 text-white'
                    : 'bg-red-600 text-white'
              }`}
            >
              {/* Trash Icon */}
              <svg
                className={`w-4 h-4 transition-all duration-300 ${
                  deletePhase === 'transforming' || deletePhase === 'fading'
                    ? 'opacity-0 scale-0 rotate-180'
                    : 'opacity-100 scale-100 rotate-0'
                }`}
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
              >
                <path
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth={2}
                  d="M19 7l-.867 12.142A2 2 0 0116.138 21H7.862a2 2 0 01-1.995-1.858L5 7m5 4v6m4-6v6m1-10V4a1 1 0 00-1-1h-4a1 1 0 00-1 1v3M4 7h16"
                />
              </svg>

              {/* Checkmark Icon */}
              <svg
                className={`absolute inset-0 m-auto w-4 h-4 text-white transition-all duration-300 ${
                  deletePhase === 'transforming'
                    ? 'opacity-100 scale-110 animate-bounce'
                    : deletePhase === 'fading'
                      ? 'opacity-100 scale-100'
                      : 'opacity-0 scale-0'
                }`}
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
              >
                <path
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth={3}
                  d="M5 13l4 4L19 7"
                />
              </svg>
            </Button>
          </div>
        </div>
        {children}
      </div>

      {isEditing && (
        <div
          className="fixed inset-0 bg-black bg-opacity-70 backdrop-blur-sm flex items-center justify-center z-50 animate-fadeIn rounded-xl !mt-[0]"
          onClick={handleCancel}
        >
          <div
            className="bg-gray-800 border border-gray-600 rounded-xl p-6 w-full max-w-md mx-4 shadow-2xl animate-slideIn"
            onClick={(e) => e.stopPropagation()}
          >
            <h3 className="text-xl font-semibold mb-6 text-gray-200 flex items-center gap-2">
              <svg
                className="w-5 h-5 text-blue-400"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
              >
                <path
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth={2}
                  d="M11 5H6a2 2 0 00-2 2v11a2 2 0 002 2h11a2 2 0 002-2v-5m-1.414-9.414a2 2 0 112.828 2.828L11.828 15H9v-2.828l8.586-8.586z"
                />
              </svg>
              Edit Task
            </h3>

            <input
              type="text"
              value={editTitle}
              onChange={(e) => setEditTitle(e.target.value)}
              className="w-full p-4 bg-gray-700 border border-gray-600 rounded-lg text-white placeholder-gray-400 focus:ring-2 focus:ring-blue-500 focus:border-transparent outline-none mb-6 transition-all duration-200"
              placeholder="Enter task title..."
              autoFocus
              onKeyDown={(e) => {
                if (e.key === 'Enter') handleSave();
                if (e.key === 'Escape') handleCancel();
              }}
            />

            <div className="flex gap-3 justify-end">
              <Button
                onClick={handleCancel}
                className="px-6 py-3 text-gray-300 bg-gray-700 hover:bg-gray-600 border border-gray-600 rounded-lg transition-all duration-200"
              >
                Cancel
              </Button>
              <Button
                onClick={handleSave}
                className="px-6 py-3 bg-gradient-to-r from-blue-600 to-blue-700 hover:from-blue-700 hover:to-blue-800 text-white rounded-lg transition-all duration-200 shadow-lg hover:shadow-xl"
              >
                Save Changes
              </Button>
            </div>
          </div>
        </div>
      )}
    </>
  );
}
