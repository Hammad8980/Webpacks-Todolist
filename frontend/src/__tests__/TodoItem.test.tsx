import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import TodoItem from '../components/todo/TodoItem';
import type { Task } from '../features/todos/TodoTaskTypes';

describe('TodoItem', () => {
  const mockOnDelete = jest.fn();
  const mockOnToggle = jest.fn();
  const mockOnUpdate = jest.fn();

  const defaultTask: Task = {
    id: 1,
    title: 'Test task',
    isCompleted: false,
    priority: 'p2',
  };

  const defaultProps = {
    task: defaultTask,
    onDelete: mockOnDelete,
    onToggle: mockOnToggle,
    onUpdate: mockOnUpdate,
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders task title and controls', () => {
    const { container } = render(<TodoItem {...defaultProps} />);

    expect(screen.getByText('Test task')).toBeInTheDocument();
    expect(screen.getByRole('checkbox')).toBeInTheDocument();

    const buttons = screen.getAllByRole('button');
    expect(buttons).toHaveLength(2); // edit and delete buttons
    expect(container).toMatchSnapshot();
  });

  it('shows completed task with line-through style', () => {
    const completedTask = { ...defaultTask, isCompleted: true };
    const { container } = render(<TodoItem {...defaultProps} task={completedTask} />);

    const taskTitle = screen.getByText('Test task');
    expect(screen.getByRole('checkbox')).toBeChecked();
    expect(taskTitle).toHaveClass('line-through');
    expect(taskTitle).toHaveClass('text-gray-400');
    expect(container).toMatchSnapshot();
  });

  it('calls onToggle when checkbox is clicked', async () => {
    const user = userEvent.setup();
    render(<TodoItem {...defaultProps} />);

    const checkbox = screen.getByRole('checkbox');
    await user.click(checkbox);

    expect(mockOnToggle).toHaveBeenCalledWith(1);
  });

  it('checkbox reflects task completion status', () => {
    const { rerender } = render(<TodoItem {...defaultProps} />);

    let checkbox = screen.getByRole('checkbox');
    expect(checkbox).not.toBeChecked();

    const completedTask = { ...defaultTask, isCompleted: true };
    rerender(<TodoItem {...defaultProps} task={completedTask} />);

    checkbox = screen.getByRole('checkbox');
    expect(checkbox).toBeChecked();
  });

  it('opens edit modal when edit button is clicked', async () => {
    const user = userEvent.setup();
    const { container } = render(<TodoItem {...defaultProps} />);

    const buttons = screen.getAllByRole('button');
    const editButton = buttons[0]; // First button is edit
    await user.click(editButton);

    expect(screen.getByText('Edit Task')).toBeInTheDocument();
    expect(screen.getByDisplayValue('Test task')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('closes edit modal when cancel button is clicked', async () => {
    const user = userEvent.setup();
    render(<TodoItem {...defaultProps} />);

    const buttons = screen.getAllByRole('button');
    const editButton = buttons[0];
    await user.click(editButton);

    const cancelButton = screen.getByRole('button', { name: /cancel/i });
    await user.click(cancelButton);

    expect(screen.queryByText('Edit Task')).not.toBeInTheDocument();
  });

  it('saves task when Enter key is pressed in edit input', async () => {
    const user = userEvent.setup();
    render(<TodoItem {...defaultProps} />);

    const buttons = screen.getAllByRole('button');
    const editButton = buttons[0];
    await user.click(editButton);

    const input = screen.getByDisplayValue('Test task');
    await user.clear(input);
    await user.type(input, 'Updated task{enter}');

    expect(mockOnUpdate).toHaveBeenCalledWith(1, 'Updated task');
    expect(screen.queryByText('Edit Task')).not.toBeInTheDocument();
  });

  it('updates task title when save button is clicked', async () => {
    const user = userEvent.setup();
    render(<TodoItem {...defaultProps} />);

    const buttons = screen.getAllByRole('button');
    const editButton = buttons[0];
    await user.click(editButton);

    const input = screen.getByDisplayValue('Test task');
    await user.clear(input);
    await user.type(input, 'Updated task');

    const saveButton = screen.getByRole('button', { name: /save changes/i });
    await user.click(saveButton);

    expect(mockOnUpdate).toHaveBeenCalledWith(1, 'Updated task');
    expect(screen.queryByText('Edit Task')).not.toBeInTheDocument();
  });

  it('does not save empty task', async () => {
    const user = userEvent.setup();
    render(<TodoItem {...defaultProps} />);

    const buttons = screen.getAllByRole('button');
    const editButton = buttons[0];
    await user.click(editButton);

    const input = screen.getByDisplayValue('Test task');
    await user.clear(input);

    const saveButton = screen.getByRole('button', { name: /save changes/i });
    await user.click(saveButton);

    expect(mockOnUpdate).not.toHaveBeenCalled();
    expect(screen.getByText('Edit Task')).toBeInTheDocument();
  });

  it('calls onDelete when delete button is clicked', async () => {
    jest.useFakeTimers();
    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });

    render(<TodoItem {...defaultProps} />);

    const buttons = screen.getAllByRole('button');
    const deleteButton = buttons[1]; // Second button is delete
    await user.click(deleteButton);

    // Advance through animation
    jest.advanceTimersByTime(1200);

    await waitFor(() => {
      expect(mockOnDelete).toHaveBeenCalledWith(1);
    });

    jest.useRealTimers();
  });
});
