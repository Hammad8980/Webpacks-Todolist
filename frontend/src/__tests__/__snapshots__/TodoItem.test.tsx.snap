// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`TodoItem opens edit modal when edit button is clicked 1`] = `
<div>
  <div
    class="relative flex items-center gap-3 p-4 rounded-xl border transition-all duration-200 shadow-sm hover:shadow-md overflow-hidden bg-gray-700 hover:bg-gray-650 border-gray-600"
  >
    <div
      class="absolute inset-0 bg-red-600 transition-all duration-500 ease-out scale-x-0 opacity-0"
      style="transform-origin: right center; border-radius: inherit;"
    />
    <div
      class="relative z-10 flex items-center gap-3 w-full"
    >
      <input
        class="h-5 w-5 text-blue-500 bg-gray-600 border-gray-500 rounded focus:ring-blue-400 focus:ring-2"
        type="checkbox"
      />
      <span
        class="flex-grow text-lg transition-colors duration-200 text-justify break-all text-gray-200 "
      >
        Test task
      </span>
      <div
        class="flex gap-2"
      >
        <button
          class="bg-blue-600 hover:bg-blue-700 text-white p-2 rounded-lg transition-all duration-200 shadow-sm hover:shadow-md disabled:opacity-50 disabled:cursor-not-allowed"
        >
          <svg
            class="w-4 h-4"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              d="M11 5H6a2 2 0 00-2 2v11a2 2 0 002 2h11a2 2 0 002-2v-5m-1.414-9.414a2 2 0 112.828 2.828L11.828 15H9v-2.828l8.586-8.586z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </button>
        <button
          class="relative p-2 rounded-lg transition-all duration-200 shadow-sm hover:shadow-md disabled:cursor-not-allowed overflow-hidden bg-red-600 hover:bg-red-700 text-white"
        >
          <svg
            class="w-4 h-4 transition-all duration-300 opacity-100 scale-100 rotate-0"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 7l-.867 12.142A2 2 0 0116.138 21H7.862a2 2 0 01-1.995-1.858L5 7m5 4v6m4-6v6m1-10V4a1 1 0 00-1-1h-4a1 1 0 00-1 1v3M4 7h16"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          <svg
            class="absolute inset-0 m-auto w-4 h-4 text-white transition-all duration-300 opacity-0 scale-0"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              d="M5 13l4 4L19 7"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="3"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
  <div
    class="fixed inset-0 bg-black bg-opacity-70 backdrop-blur-sm flex items-center justify-center z-50 animate-fadeIn rounded-xl !mt-[0]"
  >
    <div
      class="bg-gray-800 border border-gray-600 rounded-xl p-6 w-full max-w-md mx-4 shadow-2xl animate-slideIn"
    >
      <h3
        class="text-xl font-semibold mb-6 text-gray-200 flex items-center gap-2"
      >
        <svg
          class="w-5 h-5 text-blue-400"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 5H6a2 2 0 00-2 2v11a2 2 0 002 2h11a2 2 0 002-2v-5m-1.414-9.414a2 2 0 112.828 2.828L11.828 15H9v-2.828l8.586-8.586z"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
        Edit Task
      </h3>
      <input
        class="w-full p-4 bg-gray-700 border border-gray-600 rounded-lg text-white placeholder-gray-400 focus:ring-2 focus:ring-blue-500 focus:border-transparent outline-none mb-6 transition-all duration-200"
        placeholder="Enter task title..."
        type="text"
        value="Test task"
      />
      <div
        class="flex gap-3 justify-end"
      >
        <button
          class="px-6 py-3 text-gray-300 bg-gray-700 hover:bg-gray-600 border border-gray-600 rounded-lg transition-all duration-200"
        >
          Cancel
        </button>
        <button
          class="px-6 py-3 bg-gradient-to-r from-blue-600 to-blue-700 hover:from-blue-700 hover:to-blue-800 text-white rounded-lg transition-all duration-200 shadow-lg hover:shadow-xl"
        >
          Save Changes
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`TodoItem renders task title and controls 1`] = `
<div>
  <div
    class="relative flex items-center gap-3 p-4 rounded-xl border transition-all duration-200 shadow-sm hover:shadow-md overflow-hidden bg-gray-700 hover:bg-gray-650 border-gray-600"
  >
    <div
      class="absolute inset-0 bg-red-600 transition-all duration-500 ease-out scale-x-0 opacity-0"
      style="transform-origin: right center; border-radius: inherit;"
    />
    <div
      class="relative z-10 flex items-center gap-3 w-full"
    >
      <input
        class="h-5 w-5 text-blue-500 bg-gray-600 border-gray-500 rounded focus:ring-blue-400 focus:ring-2"
        type="checkbox"
      />
      <span
        class="flex-grow text-lg transition-colors duration-200 text-justify break-all text-gray-200 "
      >
        Test task
      </span>
      <div
        class="flex gap-2"
      >
        <button
          class="bg-blue-600 hover:bg-blue-700 text-white p-2 rounded-lg transition-all duration-200 shadow-sm hover:shadow-md disabled:opacity-50 disabled:cursor-not-allowed"
        >
          <svg
            class="w-4 h-4"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              d="M11 5H6a2 2 0 00-2 2v11a2 2 0 002 2h11a2 2 0 002-2v-5m-1.414-9.414a2 2 0 112.828 2.828L11.828 15H9v-2.828l8.586-8.586z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </button>
        <button
          class="relative p-2 rounded-lg transition-all duration-200 shadow-sm hover:shadow-md disabled:cursor-not-allowed overflow-hidden bg-red-600 hover:bg-red-700 text-white"
        >
          <svg
            class="w-4 h-4 transition-all duration-300 opacity-100 scale-100 rotate-0"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 7l-.867 12.142A2 2 0 0116.138 21H7.862a2 2 0 01-1.995-1.858L5 7m5 4v6m4-6v6m1-10V4a1 1 0 00-1-1h-4a1 1 0 00-1 1v3M4 7h16"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          <svg
            class="absolute inset-0 m-auto w-4 h-4 text-white transition-all duration-300 opacity-0 scale-0"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              d="M5 13l4 4L19 7"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="3"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`TodoItem shows completed task with line-through style 1`] = `
<div>
  <div
    class="relative flex items-center gap-3 p-4 rounded-xl border transition-all duration-200 shadow-sm hover:shadow-md overflow-hidden bg-gray-700 hover:bg-gray-650 border-gray-600"
  >
    <div
      class="absolute inset-0 bg-red-600 transition-all duration-500 ease-out scale-x-0 opacity-0"
      style="transform-origin: right center; border-radius: inherit;"
    />
    <div
      class="relative z-10 flex items-center gap-3 w-full"
    >
      <input
        checked=""
        class="h-5 w-5 text-blue-500 bg-gray-600 border-gray-500 rounded focus:ring-blue-400 focus:ring-2"
        type="checkbox"
      />
      <span
        class="flex-grow text-lg transition-colors duration-200 text-justify break-all line-through text-gray-400 "
      >
        Test task
      </span>
      <div
        class="flex gap-2"
      >
        <button
          class="bg-blue-600 hover:bg-blue-700 text-white p-2 rounded-lg transition-all duration-200 shadow-sm hover:shadow-md disabled:opacity-50 disabled:cursor-not-allowed"
        >
          <svg
            class="w-4 h-4"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              d="M11 5H6a2 2 0 00-2 2v11a2 2 0 002 2h11a2 2 0 002-2v-5m-1.414-9.414a2 2 0 112.828 2.828L11.828 15H9v-2.828l8.586-8.586z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </button>
        <button
          class="relative p-2 rounded-lg transition-all duration-200 shadow-sm hover:shadow-md disabled:cursor-not-allowed overflow-hidden bg-red-600 hover:bg-red-700 text-white"
        >
          <svg
            class="w-4 h-4 transition-all duration-300 opacity-100 scale-100 rotate-0"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 7l-.867 12.142A2 2 0 0116.138 21H7.862a2 2 0 01-1.995-1.858L5 7m5 4v6m4-6v6m1-10V4a1 1 0 00-1-1h-4a1 1 0 00-1 1v3M4 7h16"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          <svg
            class="absolute inset-0 m-auto w-4 h-4 text-white transition-all duration-300 opacity-0 scale-0"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              d="M5 13l4 4L19 7"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="3"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
